### ISSUE TITLE
Is it possible to implement multi task gradient boosting in xgboost?

### ISSUE THREAD

**Issue by hwu84** [dmlc/xgboost#680]
Suppose I have several related response variables Y with same features, is it possible to minimized overall loss functions from all the response variable Y so that the model can learn the relation between the response variables and make predictions for all the response variables at the same time?  Similar to the idea from multi-task neural network? 
Does it make sense?
Thanks

**Comment by khotilov** [comment]
Are you mostly interested in the multivariate/multilabel subset of multi-task learning (when all outcomes for an instance are known), or do you need to be able to deal with more general transfer learning problems (when some outcomes are not known for some instances)?

**Comment by hwu84** [comment]
@khotilov Actually both. But the first thing I want to figure out is the multivariate/multilabel subset of multi-task learning (when all outcomes for an instance are known).
It will be a nice extension when the model can also handle the situations that some outcomes are not known for some instances.
I saw #574 and it seems that multi-task learning is something we can expect in the future from xgboost?

**Comment by tqchen** [comment]
Maybe you can try to define your objective function, and use the customized gradient to achieve such feature. Let us move discussion to the roadmap issue
